source/drivers/MicroBitCompassCalibrator.cpp

```diff
@@ -226,8 +226,8 @@ CompassCalibration MicroBitCompassCalibrator::spherify(Sample3D centre, Sample3D
 Sample3D MicroBitCompassCalibrator::approximateCentre(Sample3D *data, int samples)
 {
     Sample3D c,t;
-    Sample3D centre = { 0,0,0 };
-    Sample3D best = { 0,0,0 };
+    Sample3D centre;
+    Sample3D best;
```

Comment:
If the constructor code has been reverted (was there constructor code to initialise these to 0?), shouldn't these 3 lines be reverted as well?

Reply:
There wasn't constructor code to initialise these as they are initialised to zero by default.

Reply:
@geowor01 I don't think there are any guarantees that struct fields are zero initialised unless they are placed in bss (globals).

Reply:
@jamesadevine The default constructor sets the value to zero. https://github.com/lancaster-university/microbit-dal/blob/8c2403e9643f28d19ecd852ed9916fbc090496d2/inc/types/CoordinateSystem.h#L81

Reply:
@geowor01 Ah cool :smile:, my bad!